orttraining/orttraining/test/python/perf_log/ort_module_perf_test_tools.py

```diff
@@ -0,0 +1,231 @@
+    (\
+    Model,\
+    BatchId,\
+    CommitId,\
+    ModelName,\
+    DisplayName,\
+    UseMixedPrecision,\
+    UseAutoCast,\
+    UseDeepSpeed,\
+    Optimizer,\
+    BatchSize,\
+    SeqLen,\
+    PredictionsPerSeq,\
+    NumOfBatches,\
+    WeightUpdateSteps,\
+    Round,\
+    GradAccSteps,\
+    AvgTimePerBatch,\
+    Throughput,\
+    StabilizedThroughput,\
+    EndToEndThroughput,\
+    TotalTime,\
+    AvgCPU,\
+    Memory,\
+    RunConfig,\
+    Time)\
+VALUES\
+(\
+\"{Model}\",\
+\"{BatchId}\",\
+\"{CommitId}\",\
+\"{ModelName}\",\
+\"{DisplayName}\",\
+{UseMixedPrecision},\
+{UseAutoCast},\
+{UseDeepSpeed},\
+\"{Optimizer}\",\
+{BatchSize},\
+{SeqLen},\
+{PredictionsPerSeq},\
+{NumOfBatches},\
+{WeightUpdateSteps},\
+{Round},\
+{GradAccSteps},\
+{AvgTimePerBatch},\
+{Throughput},\
+{StabilizedThroughput},\
+{EndToEndThroughput},\
+{TotalTime},\
+{AvgCPU},\
+{Memory},\
+\"{RunConfig}\",\
+\"{Time}\")";
+
+# Obtain connection string information from the portal
+def ConnectToPerfDashboardDb(mysql_server_name, power_bi_user_name, password, database):
+    config = {
+        'host': mysql_server_name,
+        'user': power_bi_user_name,
+        'password': password,
+        'database': database,
+    }
+
+    try:
+        conn = mysql.connector.connect(**config)
+        print("Connection established")
+        return conn
+    except mysql.connector.Error as err:
+        if err.errno == errorcode.ER_ACCESS_DENIED_ERROR:
+            print("Something is wrong with the user name or password")
+        elif err.errno == errorcode.ER_BAD_DB_ERROR:
+            print("Database does not exist")
+        else:
+            print(err)
+
+def log_perf_metrics(perf_metrics,
+    mysql_server_name, power_bi_user_name, power_bi_password, power_bi_database, perf_repo_path=None):
+    if perf_repo_path:
+        perf_metrics['CommitId'] = get_repo_commit(perf_repo_path)
+    else:
+        perf_metrics['CommitId'] = get_repo_commit(os.path.realpath(__file__))
+
+    ConnectAndInsertPerfMetrics(
+        mysql_server_name,
+        power_bi_user_name,
+        power_bi_password,
+        power_bi_database,
+        perf_metrics)
+
+required_attributes_for_perf_metrics = ['model_name', 'optimizer', 'batch_size', 'epochs', 'train_steps',
+    'sequence_length']
+
+def calculate_and_log_perf_metrics(args, start_time,
+    mysql_server_name, power_bi_user_name, power_bi_password, power_bi_database, ort_repo_path=None):
+    completion_time = datetime.datetime.now()
+    perf_metrics_duration = completion_time - start_time
+
+    for attribute in required_attributes_for_perf_metrics:
+        if not hasattr(args, attribute):
+            raise ValueError('args does not contain all attributes needed to calculate perf metrics. \
+                Please prepare perf_metrics and call log_perf_metrics instead')
+
+    perf_metrics = {}    
+    perf_metrics['Model'] = args.model_name
```

Comment:
Can this lead to some kind of injection attacks?

Reply:
used parameter to guard against injection attacks. also made the code more precise